contracts/zero-ex/contracts/src/features/ERC721OrdersFeature.sol

```diff
@@ -0,0 +1,1112 @@
+    ///      specify any properties, this function instead checks 
+    ///      whether the given token ID matches the ID in the order.
+    ///      Reverts if any checks fail, or if the order is selling 
+    ///      an ERC721 asset.
+    /// @param order The ERC721 order.
+    /// @param erc721TokenId The ID of the ERC721 asset.
+    function validateERC721OrderProperties(
+        LibERC721Order.ERC721Order memory order,
+        uint256 erc721TokenId
+    )
+        public
+        override
+        view
+    {
+        // Order must be selling an buying an ERC721 asset to 
+        // have properties.
+        require(
+            order.direction == LibERC721Order.TradeDirection.BUY_721,
+            "ERC721OrdersFeature::validateERC721OrderProperties/WRONG_TRADE_DIRECTION"
+        );
+
+        // If no properties are specified, check that the given 
+        // `erc721TokenId` matches the one specified in the order. 
+        if (order.erc721TokenProperties.length == 0) {
+            rrequire(
+                erc721TokenId == order.erc721TokenId,
+                LibERC721OrdersRichErrors.ERC721TokenIdMismatchError(
+                    erc721TokenId,
+                    order.erc721TokenId
+                )
+            );
+        }
+
+        // Validate each property
+        for (uint256 i = 0; i < order.erc721TokenProperties.length; i++) {
+            LibERC721Order.Property memory property = order.erc721TokenProperties[i];
+            // `address(0)` is interpreted as a no-op. Any token ID
+            // will satisfy a property with `propertyValidator == address(0)`.
+            if (address(property.propertyValidator) == address(0)) {
+                continue;
+            }
+
+            // Call the property validator and throw a descriptive error
+            // if the call reverts.
+            try property.propertyValidator.validateProperty(
+                address(order.erc721Token),
+                erc721TokenId,
+                property.propertyData
+            ) {} catch (bytes memory errorData) {
+                LibERC721OrdersRichErrors.PropertyValidationFailedError(
+                    address(property.propertyValidator),
+                    address(order.erc721Token),
+                    erc721TokenId,
+                    property.propertyData,
+                    errorData
+                ).rrevert();
+            }
+        }
+    }
+
+    /// @dev Get the current status of an ERC721 order.
+    /// @param order The ERC721 order.
+    /// @return status The status of the order.
+    function getERC721OrderStatus(LibERC721Order.ERC721Order memory order)
+        public
+        override
+        view
+        returns (LibERC721Order.OrderStatus status)
+    {
+        // Only sell orders with `erc721TokenId` == 0 can be property
+        // orders. 
+        if (order.erc721TokenProperties.length > 0 && 
+                (order.direction == LibERC721Order.TradeDirection.SELL_721 ||
```

Comment:
should this be `||` or `&&`?

Reply:
Ah yes `||` is correct but the comment is wrong; should read "Only buy orders..."